dll/win32/dbghelp/dbghelp.c

```diff
@@ -191,21 +191,68 @@ struct cpu* cpu_find(DWORD machine)
     return NULL;
 }
 
+static WCHAR* make_default_search_path(void)
+{
+    WCHAR*      search_path;
+    WCHAR*      p;
+    unsigned    sym_path_len;
+    unsigned    alt_sym_path_len;
+
+    sym_path_len = GetEnvironmentVariableW(L"_NT_SYMBOL_PATH", NULL, 0);
+    alt_sym_path_len = GetEnvironmentVariableW(L"_NT_ALT_SYMBOL_PATH", NULL, 0);
+
+    /* The default symbol path is ".[;%_NT_SYMBOL_PATH%][;%_NT_ALT_SYMBOL_PATH%]".
+     * If the variables exist, the lengths include a null-terminator. We use that
+     * space for the semicolons, and only add the initial dot and the final null. */
+    search_path = HeapAlloc(GetProcessHeap(), 0,
+                            (1 + sym_path_len + alt_sym_path_len + 1) * sizeof(WCHAR));
+    if (!search_path) return NULL;
+
+    p = search_path;
+    *p++ = L'.';
+    if (sym_path_len)
+    {
+        *p++ = L';';
+        GetEnvironmentVariableW(L"_NT_SYMBOL_PATH", p, sym_path_len);
```

Comment:
In theory there is a TOCTOU problem here, because if the env-var would be updated to be longer than `sym_path_len`, the result in the `p` buffer would be undefined according to msdn.
Not sure if this should be take into consideration at all, probably not tbh.
https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getenvironmentvariable

Reply:
Yeah, I'd expect if somebody is messing with environment variables in another thread, it's the program's own fault. Especially since dbghelp documentation specifically says that it's single-threaded.
We won't crash or anything, so I think this is probably good enough. Could potentially zero the heap allocation so that we won't get uninitialized garbage in the buffer if one of the calls fails.